lint-rules-android-lint/src/main/java/com/vanniktech/lintrules/android/MatchingViewIdDetector.kt

```diff
@@ -28,11 +28,19 @@ class MatchingViewIdDetector : LayoutDetector() {
       val fix = fix()
           .replace()
           .text(id)
-          .with(fileName)
+          .with(toBeReplaced(fileName, id))
           .autoFix()
           .build()
 
       context.report(ISSUE_MATCHING_VIEW_ID, attribute, context.getValueLocation(attribute), "Id should start with: $fileName", fix)
     }
   }
+
+  private fun toBeReplaced(fileName: String, id: String): String {
```

Comment:
Let's reuse this method and then we can merge it

Reply:
I've actually extracted it into a class for extensibility.